sam/feature_engineering/simple_feature_engineering.py

```diff
@@ -0,0 +1,198 @@
+                    )
+            else:
+                datetime = datetime.dt.tz_localize("UTC")
+            datetime = datetime.dt.tz_convert(self.timezone)
+        return datetime
+
+    def _get_time_column(self, X, component):
+        # First select the datetime column
+        if self.time_col is not None:
+            datetime = X[self.time_col]
+        else:
+            datetime = X.index.to_series().copy()
+
+        # Fix timezone
+        datetime = self._fix_timezone(datetime)
+
+        # Then select the component
+        if component in self.valid_components:
+            return self.component_function[component](datetime)
+        else:
+            raise ValueError(f"Invalid component: {component}")
+
+    def _get_rolling_features(self, X):
+        X_out = pd.DataFrame(index=X.index, columns=[])
+        # Rolling features
+        if self.time_col is not None:
+            X = X.set_index(self.time_col)
+
+        for col, method, window in self.rolling_features:
+            colname = f"{col}_{method}_{window}"
+            if method == "lag":
+                X_out[colname] = X[col].shift(window).values
+            else:
+                X_out[colname] = X[col].rolling(window=window).agg(method).values
+
+        if self.time_col is not None:
+            X = X.reset_index(drop=False)
+        return X_out
+
+    def _get_time_features(self, X):
+        X_out = pd.DataFrame(index=X.index, columns=[])
+
+        for component, type in self.time_features:
+            colname = f"{component}_{type}"
+            comp_min, comp_max = self.component_range[component]
+
+            if type == "onehot":
+                # we do not make a dummy of the last value because of collinearity
+                if self.drop_first:
+                    comp_min += 1
+                for value in range(comp_min, comp_max + 1):
+                    comp_series = self._get_time_column(X, component)
+                    colname_ = f"{colname}_{value}"
+                    X_out[colname_] = (comp_series == value).astype(int)
+
+            elif type == "cyclical":
+                comp_series = self._get_time_column(X, component)
+                # scale to 0,1, then to 0,2pi and then to -1,1
+                comp_norm = (comp_series - comp_min) / (comp_max - comp_min + 1)
+                X_out[colname + "_sin"] = np.sin(2 * np.pi * comp_norm).astype(float)
+                X_out[colname + "_cos"] = np.cos(2 * np.pi * comp_norm).astype(float)
+            else:
+                raise ValueError(f"Invalid type: {type}")
+
+        return X_out
+
+    def feature_engineer_(self, X, y=None):
+        X = X.copy()
+        if self.keep_original:
+            X_out = X.copy()
+        else:
+            X_out = pd.DataFrame(index=X.index, columns=[])
+
+        X_out = pd.concat([X_out, self._get_rolling_features(X)], axis=1)
+        X_out = pd.concat([X_out, self._get_time_features(X)], axis=1)
```

Comment:
Shouldn't we also add the imputer here? Since the previous implementation had the simpleImputer stap in here. Then again, that would be harder to overwrite when you want the iterativeimputer...